api/src/main/java/com/cloud/network/Network.java

```diff
@@ -44,6 +46,24 @@
         Shared, Isolated, L2
     }
 
+    enum PVlanType {
+        Community, Isolated, Promiscuous;
+
+        static PVlanType fromValue(String type) {
+            if (StringUtils.isBlank(type)) {
+                return null;
+            } else if (type.equalsIgnoreCase("promiscuous") || type.equalsIgnoreCase("p")) {
+                return Promiscuous;
+            } else if (type.equalsIgnoreCase("community") || type.equalsIgnoreCase("c")) {
+                return Community;
+            } else if (type.equalsIgnoreCase("isolated") || type.equalsIgnoreCase("i")) {
+                return Isolated;
+            } else {
+                throw new InvalidParameterValueException("Unexpected Private VLAN type: " + type);
```

Comment:
if we are allowing 'i', a letter which is in each of the words, I would rather have a begins with construct, allowing partial words being entered. i.e. "promiscuous".startsWith(type)...

Reply:
What about type.`length() == 1 && type.equalsIgnoreCase("i")` for the single letter case? Otherwise we are allowing substrings and we would like either the first letter or the whole type

Reply:
I wouldn't mind allowing "iso" or "Comm" in this case would you?

Reply:
Yes, I would prefer to keep only the entire type or the first letter, sorry